app/src/main/java/io/homeassistant/companion/android/sensors/NetworkSensorManager.kt

```diff
@@ -174,7 +176,25 @@ class NetworkSensorManager : SensorManager {
             conInfo = wifiManager.connectionInfo
         }
 
-        val bssid = if (conInfo!!.bssid == null) "<not connected>" else conInfo.bssid
+        var bssid = if (conInfo!!.bssid == null) "<not connected>" else conInfo.bssid
+
+        val settingName = "replace_$bssid"
+        val sensorDao = AppDatabase.getInstance(context).sensorDao()
+        val fullSensor = sensorDao.getSettings(bssidState.id)
+        val getNextBSSID = fullSensor?.settings?.firstOrNull { it.name == "get_next_bssid" }?.value ?: "false"
+        val currentSetting = fullSensor?.settings?.firstOrNull { it.name == settingName }?.value ?: ""
+        if (getNextBSSID == "true") {
+            if (currentSetting == "") {
+                sensorDao.add(Setting(bssidState.id, "get_next_bssid", "false", "toggle"))
```

Comment:
If we are still on the same BSSID on the next update should we turn it off?

Reply:
I think I should rename it to say `get_current_bssid` since thats actually what it is doing :) I thought of this the other night since while the user is on this screen we update it every 10 seconds so it would be pretty difficult for it to be the actual next BSSID in reality.